nextflow.config

```diff
@@ -9,58 +9,119 @@
-    email_on_fail              = null
-    plaintext_email            = false
-    monochrome_logs            = false
-    hook_url                   = null
-    help                       = false
-    validate_params            = true
-    show_hidden_params         = false
-    schema_ignore_params       = 'genomes'
-    enable_conda               = false
+    // Reference genome information; iGenomes is effectively disabled but retained for linting
+    genome          = null
+    force_genome    = false
+    igenomes_base   = 's3://ngi-igenomes/igenomes'
+    igenomes_ignore = true
+    hmf_genome_base = 'https://pub-cf6ba01919994c3cbd354659947f74d8.r2.dev/genomes'
+
+    // Other reference data and config exposed to user on CLI
+    max_fastq_records        = 10000000
+
+    isofox_counts            = null
+    isofox_gc_ratios         = null
+    isofox_gene_ids          = null
+    isofox_tpm_norm          = null
+    isofox_read_length       = null
+    isofox_functions         = 'TRANSCRIPT_COUNTS;ALT_SPLICE_JUNCTIONS;FUSIONS;RETAINED_INTRONS'
+
+    gridss_config            = null
+
+    prepare_reference_only   = false
+    create_stub_placeholders = false
 
+    // Boilerplate options
+    outdir           = null
+    publish_dir_mode = 'copy'
+    email            = null
+    email_on_fail    = null
+    plaintext_email  = false
+    monochrome_logs  = false
+    hook_url         = null
+    help             = false
+    version          = false
 
     // Config options
+    config_profile_name        = null
+    config_profile_description = null
     custom_config_version      = 'master'
     custom_config_base         = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
-    config_profile_description = null
     config_profile_contact     = null
     config_profile_url         = null
-    config_profile_name        = null
-
 
     // Max resource options
     // Defaults only, expecting to be overwritten
-    max_memory                 = '128.GB'
-    max_cpus                   = 16
-    max_time                   = '240.h'
+    max_memory = '128.GB'
+    max_cpus   = 16
+    max_time   = '240.h'
+
+    // Parameter lint ignore list
+    // NOTE(SW): entries here are generally have conditional defaults or are for internal use only
+    lint_ignore = [
+        'lint_ignore',
+        'genome_type',
+        'genome_version',
+        'genomes',
+        'hmf_genome_base',
+        'hmf_data_paths',
+        'panel',
+        'panel_data_paths',
+        'ref_data',
+        'ref_data_genome_alt',
+        'ref_data_genome_bwamem2_index',
+        'ref_data_genome_dict',
+        'ref_data_genome_fai',
+        'ref_data_genome_fasta',
+        'ref_data_genome_gridss_index',
+        'ref_data_genome_gtf',
+        'ref_data_genome_star_index',
+        'ref_data_hla_slice_bed',
+        'ref_data_hmf_data_path',
+        'ref_data_panel_data_path',
+        'ref_data_virusbreakenddb_path',
+    ]
```

Comment:
@mirpedrol can this not go directlz into the .nf-core.yml ?

Reply:
I did take a look at using .nf-core.yml for this but wasn't able to get it working. Though if @mirpedrol has a better solution that would be great 😁

Reply:
I'll mark this as resolved for now - @mirpedrol if you have an alternative approach, please do let us know here or elsewhere!